src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
I'm afraid I don't follow. This is `Headers.__init__()`:

```python
    def __init__(self, rawHeaders=None):
        self._rawHeaders = {}
        if rawHeaders is not None:
            for name, values in rawHeaders.items():
                self.setRawHeaders(name, values)
```

It calls `setRawHeaders()` which is [clearly documented](https://twistedmatrix.com/documents/17.1.0/api/twisted.web.http_headers.Headers.html#setRawHeaders) as accepting `bytes` or `unicode`. This enhancement was done as part of the Python 3 porting work, if I recall correctly, and it has prompted the removal of lots of not-quite-right header encoding code from `treq`. Like, in particular, the stuff that assumed headers were ASCII that prompted this PR in the first place. :-)

In any case I actually _am_ passing `bytes` here. I just wrote out the literals as `u''` because it makes the test easier to read than the equivalent backslash escapes. Plus, it keeps the lines shorter.

If you're asking, "why would a Content-Type header contain Unicode in real life", well, I hope that doesn't actually happen, but I don't see any reason that treq should produce a UnicodeDecodeError should such a thing happen. In particular, it shouldn't do that _on Python 3 but not on Python 2_.